modApiServer/src/org/aion/api/server/rpc/ApiWeb3Aion.java

```diff
@@ -2267,6 +2268,70 @@ public RpcMsg ops_getTransaction(Object _params) {
         return new RpcMsg(result);
     }
 
+    public RpcMsg ops_getBlockDetails(Object _params){
+        String _bnOrHash;
+        if (_params instanceof JSONArray) {
+            _bnOrHash = ((JSONArray) _params).get(0) + "";
+        } else if (_params instanceof JSONObject) {
+            _bnOrHash = ((JSONObject) _params).get("block") + "";
+        } else {
+            return new RpcMsg(null, RpcError.INVALID_PARAMS, "Invalid parameters");
+        }
+        AionBlock block = null;
+        Long bn = this.parseBnOrId(_bnOrHash);
+
+        // user passed a Long block number
+        if (bn != null) {
+            if (bn >= 0) {
+                block = (AionBlock) this.ac.getBlockchain().getBlockByNumber(bn);
+                if (block == null) {
+                    return new RpcMsg(JSONObject.NULL);
+                }
+            } else {
+                return new RpcMsg(JSONObject.NULL);
+            }
+        }
+
+        // see if the user passed in a hash
+        if (block == null) {
+            block = (AionBlock) this.ac.getBlockchain().getBlockByHash(ByteUtil.hexStringToBytes(_bnOrHash));
+            if (block == null) {
+                return new RpcMsg(JSONObject.NULL);
+            }
+        }
+
+        Block mainBlock = this.ac.getBlockchain().getBlockByNumber(block.getNumber());
```

Comment:
Is there a reason we shouldn't return block details for non-main-chain blocks?

Reply:
There is no reason why we shouldn't but since the change is specifically designed for the dashboard I added the check since the dashboard only displays mainchain blocks.

Reply:
Sounds reasonable